compiler/quilt/tools/util.py

```diff
@@ -81,3 +81,20 @@ def gzip_compress(data):
     with gzip.GzipFile(fileobj=buf, mode='wb') as fd:
         fd.write(data)
     return buf.getvalue()
+
+def children(path, dirs=True, files=True, noinvisible=True):
```

Comment:
Strictly speaking, this isn't 100% what we want. You can't say, "only give me directories" - only "don't give me files". So if the user is crazy enough to create devices, sockets, pipes, broken symlinks, etc., things will still break. But maybe it doesn't matter.

Reply:
Something like `(isdir(...) and dirs) or (isfile(...) and files)` should do it.